### ISSUE TITLE
Improve dependency file generation and parsing

### ISSUE THREAD

**Issue by GPMueller** [Trick-17/clang-build#50]
It turns out that multiple files can be specified per line (see e.g. [here](https://stackoverflow.com/a/5584472/4069571)), separated by whitespace, while there may appear whitespace inside a file name, which clang will escape with `\ `.
At the same time, as far as I can tell, clang does not escape backslashes in Windows paths (maybe it's double backslashes).

We need to either/or
- parse the file correctly, i.e. whitespace and `\ \n` separate file names and `\ ` whitespace and `\\` are part of file names
- use the `-MV` flag so that file names are in quotes, see [here](https://clang.llvm.org/docs/UsersManual.html#other-options)

I tried to do the first but failed, as the strings python parsed from the file, when inserted into pathlib, did not yield correct file names. Splitting the string at whitespace did not work, as the `\ ` whitespace was split as well...

**Comment by NOhs** [comment]
According to the link you mention about the `-MV` flag shouldn't whitespaces be escaped in the file?

**Comment by NOhs** [comment]
Just use

```python
import re

my_string = 'C:\\A\\B\\C.exe C:\\A\\B\\C\ and\ g.exe'
re.split(r'(?<!\\)\s+', my_string)
```

**Comment by GPMueller** [comment]
Implemented with 56ff087d76a84d5c47c99cee8e9b50327f8fe17c on `feature-improved-config`.
